tests/src/RoutesAndParams.test.js

```diff
@@ -275,4 +267,209 @@ describe('RoutesAndParams /routes-and-params/inner-html', () => {
+    expect(element).toBeTruthy();
+  });
+
+  test('relative urls redirect', async () => {
+    await Promise.all([
+       page.click('[href="relative"]'),
+       page.waitForNavigation(),
+    ])
+    const url = await page.url();
+    expect(url).toEqual('http://localhost:6969/routes-and-params/relative');
+  });
+  
+  test('relative composed urls redirect', async () => {
+    await Promise.all([
+       page.click('[href="relative/6969"]'),
+       page.waitForNavigation(),
+    ])
+    const url = await page.url();
+    expect(url).toEqual('http://localhost:6969/routes-and-params/relative/6969');
+  });
+
+  test('root urls redirect', async () => {
+    await Promise.all([
+       page.click('[href="/root/6969"]'),
+       page.waitForNavigation(),
+    ])
+    const url = await page.url();
+    expect(url).toEqual('http://localhost:6969/root/6969');
+  });
+
+  test('current urls redirect', async () => {
+    await Promise.all([
+       page.click('[href="./current"]'),
+       page.waitForNavigation(),
+    ])
+    const url = await page.url();
+    expect(url).toEqual('http://localhost:6969/routes-and-params/current');
+  });
+
+  test('parent urls redirect', async () => {
+    await Promise.all([
+       page.click('[href="../parent"]'),
+       page.waitForNavigation(),
+    ])
+    const url = await page.url();
+    expect(url).toEqual('http://localhost:6969/parent');
+  });
+
+});
+
+
+describe('ssr RoutesAndParams /routes-and-params/hrefs', () => {
+
+  test('ssr parent', async () => {
+    await Promise.all([
+      page.goto('http://localhost:6969/routes-and-params/hrefs?parent=1'),
+      page.waitForNavigation(),
+    ]);
+    const url = await page.url();
+    expect(url).toEqual('http://localhost:6969/parent');
+  });
+  
+  test('ssr current', async () => {
+    await Promise.all([
+      page.goto('http://localhost:6969/routes-and-params/hrefs?current=1'),
+      page.waitForNavigation(),
+    ]);
+    const url = await page.url();
+    expect(url).toEqual('http://localhost:6969/routes-and-params/current');
+  });
+  
+  test('ssr https', async () => {
+    await Promise.all([
+      page.goto('http://localhost:6969/routes-and-params/hrefs?https=1'),
+      page.waitForNavigation(),
+    ]);
+    const url = await page.url();
+    expect(url).toEqual('https://nullstack.app/');
+  });
+
+  test('ssr http', async () => {
+    await Promise.all([
+      page.goto('http://localhost:6969/routes-and-params/hrefs?http=1'),
+      page.waitForNavigation(),
+    ]);
+    const url = await page.url();
+    expect(url).toEqual('http://localhost:6969/http');
+  });
+  
+  test('ssr same', async () => {
+    await Promise.all([
+      page.goto('http://localhost:6969/routes-and-params/hrefs?same=1'),
+      page.waitForNavigation(),
+    ]);
+    const url = await page.url();
+    expect(url).toEqual('http://localhost:6969/routes-and-params/href-same-protocol');
+  });
+  
+  // test.skip('ssr ftp', async () => {
```

Comment:
can we test if page.status (from nullstack not puppeteer) === 500? does it raise an error?